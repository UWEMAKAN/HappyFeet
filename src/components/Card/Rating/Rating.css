.Rating {
  align-self: flex-start;
  padding: .5rem;
  font-size: 1.4rem;
  font-weight: 700;
  background-color: #209CEE;
  color: #fff;
  display: flex;
  flex-direction: row;
  margin-bottom: 1rem;
  justify-content: center;
}

.Rating span {
  font-size: 1.3rem;
  margin-right: .2rem;
}

