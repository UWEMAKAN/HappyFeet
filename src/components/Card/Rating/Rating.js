import React from 'react';
import { AiFillStar } from 'react-icons/ai';
import propTypes from 'prop-types';

import './Rating.css';

const Rating = ({ rating }) => (
  <div className="Rating">
    <span>{rating}</span>
    <AiFillStar />
  </div>
);

Rating.propTypes = {
  rating: propTypes.number.isRequired
};

export default Rating;
