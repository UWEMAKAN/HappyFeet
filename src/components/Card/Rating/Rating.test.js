import React from 'react';
import { shallow } from 'enzyme';
import Rating from './Rating';

const renderRating = (args) => {
  const defaultProps = {
    rating: 0
  };
  const props = { ...defaultProps, ...args };
  return shallow(<Rating rating={props.rating} />);
};

describe('Rating Component', () => {
  test('should render Rating Component', () => {
    const rating = 4.5;
    const wrapper = renderRating({ rating });
    expect.assertions(2);
    expect(wrapper.find('span').text()).toEqual('4.5');
    expect(wrapper).toMatchSnapshot();
  });
});
