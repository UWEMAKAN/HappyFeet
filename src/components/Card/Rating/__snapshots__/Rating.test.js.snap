// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating Component should render Rating Component 1`] = `ShallowWrapper {}`;
